### PR TITLE
Fix usePress with HTML input elements

### PR DESCRIPTION
**Pull request by devongovett**
This makes usePress work better with HTML `<input>` elements, which have default keyboard behavior. For example, checkboxes toggle when the space key is pressed. Previously this was [somewhat broken](https://codesandbox.io/s/musing-architecture-r1v5dx?file=/src/App.js), in that onPressStart would only be triggered on key up due to the browser firing a virtual onClick event.

In #3380, I changed usePress to look at the currentTarget rather than target when determining whether to handle an event. This caused TableView's select all checkbox to stop working, because the event would bubble to the cell and get handled there (with preventDefault).

Now we allow the space key for checkboxes and other input elements to trigger onPressStart on key down, but do not call `preventDefault`, allowing the browser to toggle the checkbox.

### CODE REVIEW

**Comment by devongovett** [comment]
One slight change is that the Enter key on checkboxes will now bubble since the checkbox doesn't handle it, so table row selection will toggle in that case as well. I think this is ok?

File: packages/@react-aria/interactions/src/usePress.ts [hunk 1]
```diff
@@ -774,8 +773,35 @@ function shouldPreventDefault(target: Element) {
...
+  'checkbox',
+  'radio',
+  'range',
+  'color',
+  'file',
+  'image',
+  'button',
+  'submit',
+  'reset'
+]);
```
Comment (LFDanLu) [inline packages/@react-aria/interactions/src/usePress.ts:798]: How was this list generated? I assume by going through a list of input types and adding any that wasn't a text input?

**Review by LFDanLu — APPROVED** [review]
Did some basic testing, confirmed table select all checkbox is working now. Other components seem to work still (form submit/reset buttons, checkboxes, etc), deeper sweep to come tmrw

File: packages/@react-aria/interactions/src/usePress.ts (hunk 1 above)
Comment (devongovett) [reply packages/@react-aria/interactions/src/usePress.ts:798]: yeah https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input

**Review by reidbarber — APPROVED** [review]

**Pull request merged**